### PR TITLE
[MOB-10141] make lint rules stricter

### PR DESCRIPTION
**Pull request by lposen**
## 🔹 JIRA Ticket(s) if any

* [MOB-10141](https://iterable.atlassian.net/browse/MOB-10141)

## ✏️ Description

Made lint rules stricter, and made the appropriate code changes to reflect this.

[MOB-10141]: https://iterable.atlassian.net/browse/MOB-10141?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by evantk91 — COMMENTED** [review]
Some comments and nit picks but otherwise looking good

File: src/core/classes/IterableConfig.ts [hunk 1]
```diff
@@ -130,9 +129,9 @@ export class IterableConfig {
    * Android only feature: This controls whether the SDK should enforce encryption for all PII stored on disk.
    * By default, the SDK will not enforce encryption and may fallback to unencrypted storage in case the encryption fails.
    */
-  encryptionEnforced: boolean = false;
+  encryptionEnforced = false;
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/core/classes/IterableConfig.ts:132]: You can only do this if it's a primitive type (bool, string etc), so unfortunately we can't do that.

File: src/inApp/classes/IterableInAppMessage.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export class IterableInAppMessage {
     expiresAt: Date | undefined,
     saveToInbox: boolean,
     inboxMetadata: IterableInboxMetadata | undefined,
-    customPayload: any | undefined,
+    customPayload: unknown | undefined,
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/inApp/classes/IterableInAppMessage.ts:70]: Thanks!

File: src/inApp/classes/IterableInAppTrigger.ts [hunk 3]
```diff
@@ -8,10 +8,9 @@ export class IterableInAppTrigger {
...
 
-  static fromDict(dict: any): IterableInAppTrigger {
-    const type = dict.type as
-      | IterableInAppTriggerType
-      | IterableInAppTriggerType.immediate;
-    return new IterableInAppTrigger(type);
+  static fromDict(dict: {
+    type: IterableInAppTriggerType;
+  }): IterableInAppTrigger {
+    return new IterableInAppTrigger(dict.type);
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/inApp/classes/IterableInAppTrigger.ts:14]: Agreed!  I'll make a ticket.

File: src/core/classes/Iterable.ts [hunk 4]
```diff
@@ -458,10 +460,13 @@ export class Iterable {
    * overwrite their counterparts that already exist on the user's profile.
    * Otherwise, they are added.
    *
-   * @param {any} dataFields data fields to store in user profile
+   * @param {unknown | undefined} dataFields data fields to store in user profile
    * @param {boolean} mergeNestedObjects flag indicating whether to merge top-level objects
    */
-  static updateUser(dataFields: any, mergeNestedObjects: boolean) {
+  static updateUser(
+    dataFields: unknown | undefined,
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/core/classes/Iterable.ts:467]: Add a ticket for delving into this type -- should it be `Record<string, unknown>`?

**Review by evantk91 — APPROVED** [review]

File: src/inbox/components/IterableInboxMessageDisplay.tsx [hunk 5]
```diff
@@ -210,9 +196,9 @@ export const IterableInboxMessageDisplay = ({
   }
 
   return (
-    <View style={messageDisplayContainer}>
-      <View style={header}>
-        <View style={returnButtonContainer}>
+    <View style={styles.messageDisplayContainer}>
+      <View style={styles.header}>
+        <View style={styles.returnButtonContainer}>
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/inbox/components/IterableInboxMessageDisplay.tsx:201]: Yeah -- less lines of code, and easier to see their source.

File: src/core/classes/Iterable.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (lposen) [reply src/core/classes/Iterable.ts:467]: https://iterable.atlassian.net/browse/MOB-10416

File: .eslintrc.js [hunk 6]
```diff
@@ -21,10 +25,28 @@ module.exports = {
...
+
+      parserOptions: {
+        project: ['./tsconfig.json'], // Specify it only for TypeScript files
+        // or `project: true` in typescript-eslint version >= 5.52.0
+      },
+      rules: {
+        '@typescript-eslint/no-var-requires': ['error', { allow: ['/package\\.json$'] }],
+        '@typescript-eslint/no-require-imports': ['error', { allow: ['/package\\.json$'] }],
+      },
+    },
```
[the remark this answers is not in the document]
Comment (lposen) [reply .eslintrc.js:37]: Just making ts rules in eslint stricter

File: example/src/types/navigation.ts [hunk 7]
```diff
@@ -27,8 +27,4 @@ export type MainScreenProps<T extends keyof MainScreenParamList> =
 export type RootStackScreenProps<T extends keyof RootStackParamList> =
   StackScreenProps<RootStackParamList, T>;
 
-declare global {
-  namespace ReactNavigation {
-    interface RootParamList extends RootStackParamList {}
-  }
-}
+export type { RootStackParamList as RootParamList };
```
[the remark this answers is not in the document]
Comment (lposen) [reply example/src/types/navigation.ts:30]: This is the rule I was fixing: https://typescript-eslint.io/rules/no-namespace/

**Pull request merged**